src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/VsLangProjectProperties.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Microsoft.VisualStudio.Threading;
+using VSLangProj;
+using VSLangProj110;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties
+{
+    /// <summary>
+    /// See <see cref="VsLangProjectPropertiesProvider"/> for more info.
+    /// </summary>
+    internal partial class VsLangProjectProperties : VSLangProj.ProjectProperties
+    {
+        private readonly VSProject _vsProject;
+        private readonly IProjectThreadingService _threadingService;
+        private readonly ActiveConfiguredProject<ProjectProperties> _projectProperties;
+
+        public VsLangProjectProperties(
+            VSProject vsProject,
+            IProjectThreadingService threadingService,
+            ActiveConfiguredProject<ProjectProperties> projectProperties)
+        {
+            Requires.NotNull(vsProject, nameof(vsProject));
+            Requires.NotNull(threadingService, nameof(threadingService));
+            Requires.NotNull(projectProperties, nameof(projectProperties));
+
+            _vsProject = vsProject;
+            _threadingService = threadingService;
+            _projectProperties = projectProperties;
+        }
+
+        private ProjectProperties ProjectProperties
+        {
+            get { return _projectProperties.Value; }
+        }
+
+        private JoinableTaskFactory JoinableTaskFactory
+        {
+            get { return _threadingService.JoinableTaskFactory; }
+        }
+
+        private void RunFuncTaskSynchronously<T>(Func<T, Task> asyncActionT, T value)
+        {
+            JoinableTaskFactory.Run(
+                async () =>
+                {
+                    await asyncActionT(value).ConfigureAwait(false);
+                });
+        }
+
+        public prjOutputTypeEx OutputTypeEx
+        {
+            get
+            {
+                var configurationGeneralBrowseObjectProperties = JoinableTaskFactory.Run(ProjectProperties.GetConfigurationGeneralBrowseObjectPropertiesAsync);
+                var value = (IEnumValue)JoinableTaskFactory.Run(configurationGeneralBrowseObjectProperties.OutputTypeEx.GetValueAsync);
```

Comment:
That sounds right but I don't remember the details. You'll have to look at history.